src/dotnet/commands/dotnet-add-p2p/Program.cs

```diff
@@ -0,0 +1,174 @@
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.DotNet.Cli.CommandLine;
+using Microsoft.DotNet.Cli.Utils;
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+
+namespace Microsoft.DotNet.Tools.Add.ProjectToProjectReference
+{
+    public class AddProjectToProjectReferenceCommand
+    {
+        public static int Run(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            CommandLineApplication app = new CommandLineApplication(throwOnUnexpectedArg: false)
+            {
+                Name = "dotnet add p2p",
+                FullName = ".NET Add Project to Project (p2p) reference Command",
+                Description = "Command to add project to project (p2p) reference",
+                AllowArgumentSeparator = true,
+                ArgumentSeparatorHelpText = "Project to project references to add"
+            };
+
+            app.HelpOption("-h|--help");
+
+            CommandArgument projectArgument = app.Argument("<PROJECT>",
+                "The project file to modify. If a project file is not specified," +
+                " it searches the current working directory for an MSBuild file that has a file extension that ends in `proj` and uses that file.");
+
+            CommandOption frameworkOption = app.Option("-f|--framework <FRAMEWORK>", "Add reference only when targetting a specific framework", CommandOptionType.SingleValue);
+
+            app.OnExecute(() => {
+                if (projectArgument.Value == null)
+                {
+                    throw new GracefulException("Argument <Project> is required.");
+                }
+
+                ProjectRootElement project = File.Exists(projectArgument.Value) ?
+                                                GetProjectFromFileOrThrow(projectArgument.Value) :
+                                                GetProjectFromDirectoryOrThrow(projectArgument.Value);
+
+                if (app.RemainingArguments.Count == 0)
+                {
+                    throw new GracefulException("You must specify at least one reference to add.");
+                }
+
+                int numberOfAddedReferences = AddProjectToProjectReference(project, frameworkOption.Value(), app.RemainingArguments);
+
+                if (numberOfAddedReferences != 0)
+                {
+                    project.Save();
+                }
+
+                return 0;
+            });
+
+            try
+            {
+                return app.Execute(args);
+            }
+            catch (GracefulException e)
+            {
+                Reporter.Error.WriteLine(e.Message.Red());
+                app.ShowHelp();
+                return 1;
+            }
+        }
+
+        // There is ProjectRootElement.TryOpen but it does not work as expected
```

Comment:
Have you reached out to the MSBuild folks to figure out why that is, instead of writing this? Either the API is broken and it needs to be fixed or we are using it wrong.